woocommerce/woocommerce-android
# Adapt DateUtils to support nullable error handling

Summary
==========
Fixes https://github.com/woocommerce/woocommerce-android/issues/2581 and cover a wide range of crashes that often happen when a Woo Store site returns any slightly different date pattern. Right now our app crashes with an `IllegalArgumentException` if there's any kind of date parsing issue, since this is a very easy kind of bug to be caused by a misconfigured plugin, it would be desirable to report the issue to Sentry but avoid interrupting the user experience, so this PR replaces all exceptions throwing with a Sentry report and a WooLog input.

How to Test
==========
Verify if the date information is properly presented in the following app Views:

1. My Store time period range and stats date for each tab (day, week, month and year)
2. Order Details Shipment, Order Status and Orders List
3. Review date below reviewer name


Update release notes:
